allennlp/models/model.py

```diff
@@ -221,7 +233,10 @@ def load(cls,
 
         # Load vocabulary from file
         vocab_dir = os.path.join(serialization_dir, 'vocabulary')
-        vocab = Vocabulary.from_files(vocab_dir)
+        if os.path.exists(vocab_dir):
```

Comment:
The Ensemble classes don't need a vocabulary.  I don't particularly like how I'm handling this option here, but the other changes I come up with require many more changes to our APIs.  Suggestions very welcome ;-)

Reply:
I could require Ensembles to have a vocabulary--and then just check to verify that it matches the vocabulary in the submodels.

Reply:
Can you just override `load` on `Ensemble`?

Reply:
I don't think so--this is a `classmethod` called via `Model.load`.

Yikes--note that GitHub makes it extremely confusing what method this change is actually in.  I had to unfold a few times.

Reply:
This probably means changing all of the places we call `Model.load` to a `load_model` method, or similar, but that's probably preferable to trying to make `Model.load` also handle ensembles.

Reply:
I'm not quite following what you're thinking of.  Whether the syntax is `Model.load` or `load_model` doesn't make much difference.

The challenge is that `Model.load` makes some decisions (such as loading a Vocabulary) before it knows which class it's loading (which is specified in a parameter file).  I could peak at the model type in the parameters file and make decisions based on that.

Reply:
Yes, that's what I'm suggesting.  `load_model` would do some kind of check to figure out if it should call `Model.load()` or `Ensemble.load()`.

Reply:
Great--I'll try something like that but I don't see any need to rename the method from `Model.load()` since an `Ensemble` is a `Model`.

Reply:
What I meant:

```python
def load_archive(...):
    # We currently have something like this
    Model.load(...)
    # Instead we'd have something like this:
    load_model(...)  # note that this is not a method on `Model`

def load_model(...):
    if some_check_on_the_parameters():
        return Ensemble.load()
    else:
        return Model.load()
```

If you just modify `Model.load()`, you're going to have to break out of that somehow and call to `Ensemble.load()`, which seems pretty ugly.

Reply:
Branching out from `load_model()` to `Model.load()` (which should load any model, but in this case only loads models which are not ensembles) and `Ensemble.load()` also seems pretty ugly.

Adding another class to the hierarchy `SingleModel` (or some other name) is the only way I can see out of this.  Do you think such a class would be a good addition?

Reply:
Another option:

```python

class Model:
    def load(cls, ...):
        model_type = params.pop("type")
        return cls.by_name(model_type)._load(...)

    def _load(cls, ...):
        # The old `load()` logic here.
```

This second option is probably better, and more closely mimics what we already do with `from_params`.

Reply:
Python never ceases to amaze me.